### DISCUSSION TITLE
Automation/Script Editors: Drag and drop support

### DISCUSSION THREAD

**Discussion by niemyjski** [home-assistant/frontend#7813] [Automations / scripts / scenes]
It would be extremely useful to be able to have drag and drop support for the automation editors. Here are two scenarios this would improve:

1. Being able to logically sort conditions or triggers based on the device via sort.
2. Have an existing condition and then realizing you need to change it to an or condition, but now you have to rewrite it all. If we had drag and drop you could create a new node and then just drag the existing conditions into the proper slot.

**Comment by KTibow** [comment]
BTW you can always use YAML mode.

**Reply by niemyjski** [reply]
Yes, that's what I do now but I know drag and drop is being worked on and I think the most important thing in any api / ui is consistency and this would have really saved me some time and would be good for a lot of scenarios (noted above) especially for people who don't feel comfortable in yaml (which is more than likely a huge percentage of users).

**Reply by itsadamslife** [reply]
I frequently get frustrated with myself because I realize after I start adding conditions that I needed an 'and' or 'or' or 'not' in the condition. I usually open up the YAML to cut and paste the code in the correct spot, but it would be more user-friendly to be able to drag and drop between different sections.

**Comment by karwosts** [comment]
It's not drag-and-drop, but some of these usecases should be helped by the new UI copy/paste feature in 2023.6

**Reply by KTibow** [reply]
Drag and drop is in fact already implemented. It's more of reordering than the depth-changing they were talking about but still. Just choose in the overflow menu the reorder option.
